s3fs/core.py

```diff
@@ -146,10 +201,23 @@ def __init__(self, anon=False, key=None, secret=None, token=None,
         self.config_kwargs = config_kwargs
         self.dirs = {}
         self.req_kw = {'RequestPayer': 'requester'} if requester_pays else {}
+        self.writer_kwargs = writer_kwargs or {}
         self.use_ssl = use_ssl
         self.s3 = self.connect()
+        self._kwargs_helper = ParamKwargsHelper(self.s3)
         S3FileSystem._singleton[0] = self
 
+    def _filter_kwargs(self, s3_method, kwargs):
+        return self._kwargs_helper.filter_dict(s3_method.__name__, kwargs)
+
+    def _call_s3(self, method, *akwarglist, **kwargs):
```

Comment:
So right now this is only used for the writing methods but will work for everything